pkg/platform/common/cloudletssh/sshkey_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 EdgeXR, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudletssh
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/edgexr/edge-cloud-platform/pkg/log"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+)
+
+type keySigner struct {
+	signer      ssh.MultiAlgorithmSigner
+	signedCount int
+	validDur    time.Duration
+}
```

Comment:
I guess our primary signer would be vault-based. Are you thinking to implement that in a later PR?

Reply:
platform.AccessApi already implements the KeySigner interface from sshkey.go, so we already have a Vault-based implementation. The test code here is just emulating what Vault does internally to sign a cert.